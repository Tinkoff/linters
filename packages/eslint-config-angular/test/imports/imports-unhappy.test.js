import ESlint from 'eslint';
import path from 'path';

describe('imports / unhappy path', () => {
  const cli = new ESlint.CLIEngine({
    cwd: path.join(__dirname, '..'),
    useEslintrc: false,
    baseConfig: {
      extends: ['../internal/base', '../internal/import'],
    },
  });

  it('unhappy', () => {
    const codeframe = cli.getFormatter('codeframe');
    const report = cli.executeOnFiles([
      path.join(__dirname, './__fixtures__/imports-unhappy.fixture.ts'),
    ]);

    expect(codeframe(report.results)).toMatchSnapshot();
  });
});
