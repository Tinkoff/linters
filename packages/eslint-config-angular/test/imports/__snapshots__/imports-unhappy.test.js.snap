// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports / unhappy path unhappy 1`] = `
"error: Run autofix to sort these imports! (simple-import-sort/imports) at packages/eslint-config-angular/test/imports/__fixtures__/imports-unhappy.fixture.ts:1:1:
> 1 | import { rules as _TSLint } from '@typescript-eslint/eslint-plugin';
    | ^
  2 | import { process as _process } from 'babel-jest';
  3 | import { _A as A, _B as B } from './shared-imports.fixture';
  4 | 


1 error found.
1 error potentially fixable with the \`--fix\` option."
`;
